### ISSUE TITLE
Feature request: "--stop" CLI command should still send buildscan info to gradle enterprise

### ISSUE THREAD

**Issue by kenyee** [gradle/gradle#19010]
### Expected Behavior
Would be nice if there were a way to send a build scan to gradle enterprise when a gradle build is taking too long (we've implemented a timeout to kill the process) and is killed via "./gradlew --stop".  

### Current Behavior
It currently just ends with 
```
FAILURE: Build failed with an exception.

* What went wrong:
Gradle build daemon has been stopped: stop command received
```

### Context
We'd like to be able to flag builds that are taking too long...sometimes developers seem to have gradle daemons that have a memory leak in plugins or something that causes them to run out of memory and keep doing GC.  For these builds, we've impelmented a timeout mechanism by using a command line wrapper, but we'd like to be able to track these failures in gradle enterprise.
Not sure if Gradle is in a state that can send a build scan if it's OOM though...

**Comment by runningcode** [comment]
Can you confirm that these do not generate build scans?
Do you see any failed builds in your failures dashboard with the failure message `*Build cancelled while executing task*`?
Note that when the Gradle process deadlocks, there is no way to generate a build scan but for other cases it should generate a scan.

**Comment by kenyee** [comment]
Yep, nothing in gradle enterprise for my username.  When I do "./gradlew --stop", the running gradle run doesn't print out "your buildscan is at xxxx" either.  It just says this:
```
FAILURE: Build failed with an exception.

* What went wrong:
Gradle build daemon has been stopped: stop command received

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

```

**Comment by runningcode** [comment]
Can you search for `*Gradle build daemon has been stopped*` instead?

**Comment by kenyee** [comment]
<img width="1379" alt="Screen Shot 2021-11-30 at 10 19 19 AM" src="https://user-images.githubusercontent.com/586670/144074876-8d0ed405-90f3-400b-bd92-55dc80472922.png">

**Comment by runningcode** [comment]
It is not a custom value (unless you are tagging builds with this custom value). Can you search in your failures dashboard for this failure message pattern?

**Comment by kenyee** [comment]
there's no full text search in gradle enterprise :-(
Looked at failures for today and there's nothing at the bottom of that screen
<img width="1366" alt="Screen Shot 2021-11-30 at 12 02 45 PM" src="https://user-images.githubusercontent.com/586670/144093365-ae832f72-2763-404b-be24-aef8051e9f55.png">
:

**Comment by runningcode** [comment]
You can type your search in here:

![Screen Shot 2021-11-30 at 8 11 08 PM](https://user-images.githubusercontent.com/332597/144112280-63121ee9-081c-41d8-ab5b-58b97c0b8d76.png)

The screenshot you shared with me shows that some cancelled builds already have build scans uploaded to your Gradle Enterprise dashboard. `Could not build 'org.jetbrains.kotlin.gradle.KotlinMPPGradleModel' model. Build Cancelled.`

**Comment by kenyee** [comment]
🤦 I missed that search field....eyes were on the other fields above that you could click the refresh button on.  😹 

Anyways, same result...did
./gradlew assembleDebug
waited a bit until it got to trying to process some of the modules, then ran "./gradlew --stop" in another window.
Then hit the refresh button on that gradle enterprise screen. with "*Build cancelled while executing task*" in that other search field and no new build scans showed up...

**Comment by CRogers** [comment]
This would be incredibly useful for us (I'm on the Gradle team of a build company that uses Gradle in 1000s of projects). We sometimes have builds randomly taking 1hr+ that then do not send a buildscan - it's very hard to diagnose what's causing it to be slow.

**Comment by ov7a** [comment]
Sorry for the late reply.

This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.

**Comment by wolfs** [comment]
Gradle is trying its best to send a build scan even when using `--stop`. 

@kenyee I see that you managed to reproduce the behavior [in one of your comments earlier](https://github.com/gradle/gradle/issues/19010#issuecomment-983009261). Do you have project where I could reproduce the behavior? E.g. does running `assembleDebug` and then `gradle --stop` cause the not publishing of a build scan on e.g. the [now in Android](https://github.com/gradle/nowinandroid) project?

**Comment by kenyee** [comment]
Sorry...this is a while back and no longer work at the place that had that issue so no longer have access to a repo w/ a repro...

**Comment by ov7a** [comment]
Given the feedback, let's close this as unreproducible.

If anyone stumbles across a similar issue, please open a new one with a reproducer.
